plugins/libplugin-pay.c

```diff
@@ -457,8 +471,19 @@ static void payment_compute_onion_payloads(struct payment *p)
 			    p->start_block + p->route[i + 1].delay;
 			break;
 		case ROUTE_HOP_TLV:
-			/* TODO(cdecker) Implement */
-			abort();
+			cur->tlv_payload = tlv_tlv_payload_new(cr->hops);
+			tlvstream_set_tu64(
+			    &cur->tlv_payload->fields,
+			    TLV_TLV_PAYLOAD_AMT_TO_FORWARD,
+			    p->route[i + 1].amount.millisatoshis); /* Raw: TLV payload generation*/
+			tlvstream_set_tu32(&cur->tlv_payload->fields,
+					   TLV_TLV_PAYLOAD_OUTGOING_CLTV_VALUE,
+					   p->start_block +
+					       p->route[i + 1].delay);
+			tlvstream_set_short_channel_id(
+			    &cur->tlv_payload->fields,
+			    TLV_TLV_PAYLOAD_SHORT_CHANNEL_ID,
+			    &p->route[i + 1].channel_id);
 		}
```

Comment:
Can we generate this?  This is very unpretty... :(

Reply:
I could move it into a separate function for starters, not truly generated, but should clean this up nicely.